vignettes/user-guide.Rmd

```diff
@@ -0,0 +1,413 @@
+
+```{r include=TRUE, echo=FALSE, eval=TRUE}
+task <- ParameterIdentification$new(
+  simulations = simulations,
+  parameters = parameters,
+  outputMappings = outputMappings,
+  configuration = piConfiguration
+)
+taskResults <- task$run()
+```
+
+### Diagnostics
+
+The resulting object contains:
+
+-   `taskResults$par`: a vector of point estimates for each of the parameters
+-   `taskResults$lwr` and `taskResults$upr`: vectors with lower bounds and upper bounds for the 95% confidence interval for each of the parameters
+-   `taskResults$cv`: a vector with coefficient of variation (standard deviation over point estimate, in percents) for each of the parameters
+-   `taskResults$value` is the objective function value at the point estimate
+-   `taskResults$elapsed` (seconds) is the wall time it took to run the main optimization routine, excluding the hessian calculations
+-   `taskResults$nrOfFnEvaluations` is the number of times the objective function was evaluated in the main optimization routine, excluding the hessian calculations
+
+The `task` object also contains a `$plotResults()` function that produces three diagnostic plots: an individual time profile with the current best estimate of model parameters and overlayed observed data, a comparison of predicted-vs-observed values and a plot of residuals vs time to identify potential biases in the model. The `$plotResults()` function can take an optional vector of parameter values to plot the curve corresponding to another set of model parameters. After running the parameter identification task, the `$plotResults()` function can be called as follows:
+
+```{r include=TRUE, eval=TRUE, fig.width=10, fig.height=6, fig.align='center', warning=FALSE}
+task$plotResults()
+```
+
+We can judge if the model is properly calibrated by observing if:
+
+-   the individual time profile is close to the observed data points
+-   the predicted-vs-observed plot is close to the diagonal
+-   the residuals are randomly distributed above and below zero
+
+The residuals only above or only below zero indicate an overprediction or an underprediction. It is likely that another set of parameters will result in a better fit. A bad prediction in the early-time region of the plot indicates that the model needs a better description of the drug absorption. A bad prediction in the late-time region indicates that the model needs a better description of the drug elimination.
+
+### Plotting OFV profiles
+
+Around the global minimum, the objective function is expected to be convex and parabolic. We can check if we are, indeed, around a minimum (global or local) by plotting the objective function value (OFV) profile around the tentative optimal point. For example, after running the parameter identification task, run to plot the OFV profile:
```

Comment:
Please add more explanation for the `calculateOFVProfiles` method. Examples with lower/upper. What is the output of the method, it's structure?